Orchestrator.py

```diff
@@ -191,25 +205,81 @@ def receive(self,frame):
         # simulation complete when there are no more frontier cells left
         if not self.cellsFrontier:
             self.simEngine.completeRun()
+
         log.debug(f'remaining frontiers are {self.cellsFrontier}')
 
         # update dotBotsView
-        dotBot['x']       = newX
-        dotBot['y']       = newY
-        log.debug(f'dotbot {dotBot} is at ( {newX},{newY} ) ')
+        dotBot['x']      = newX
+        dotBot['y']      = newY
+        dotBot['bumped'] = frame['bumped']
+
+        log.debug(f'dotbot {dotBot} is at ({newX},{newY})')
+
+        # if dotBot bumped
+        if frame['bumped']:
+            self.isDirectPathOpen = False
+
+        # assign target cell (destination)
+        if not dotBot['targetCell'] or dotBot['targetCell'] not in self.cellsFrontier:
+            # target has successfully been explored
+            # assign a new target cell to dotBot
+            targetCell           = self._computeTargetCell(frame['source'])
+
+        else:
+            # target cell not explored yet, keep moving towards it
+            targetCell           = dotBot['targetCell']
+
+        # find path to target
 
-        # pick new speed and heading for dotBot
-        (heading, speed)          = self._pickNewMovement(frame['source'])
+        if not cellsExplored['cellsExplored']:
```

Comment:
I don't understand how there is a "cellsExplored" item in variable "cellsExplored"

Reply:
it computes cells explored and next cell that would be explored had robot kept moving in same direction. should i rename function to computeCellsExploredAndNextCell()?